Quality_declaration_fastcdr.md

```diff
@@ -0,0 +1,116 @@
+
+As stated in their public repository, *eProsima FastCDR is a C++ library that provides two serialization mechanisms. One is the standard CDR serialization mechanism, while the other is a faster implementation that modifies the standard*.
+
+## Summary
+The `fast-cdr` library is used as one of the dependencies of the `fastrtps`, it is important for the eProsima team to consider this software to be reliable. On its current state, meets most of the requirements to qualify as a level 1 package.
+
+As for the missing topics, eProsima does not state their versioning policies for the repository and does not state explicitly the API/ABI policies for the code in  `fast-cdr`. As the code deals with the standard RPTS implementation, and it is used by their software `fastrtps` it is not expected that the basic API functionalities will be changed. In any case, the code is actively being tested against their  [CI](http://jenkins.eprosima.com:8080/) (both directly with the included test cases and indirectly through the tests of `fastrtps`)  and indirectly in the [ROS2 CI](https://ci.ros2.org/).
+
+In terms of ROS2 package metrics this library is considered to be Quality Level 4. Adding unit testing for the functions used in ROS2 packages, coverage statistics and version pinning will be needed to achieve Quality Level 1.
+
+# Comparison with ROS packages quality standards
+
+## Version Policy [1]
+
+### Version Scheme [1.i]
+
+There is no public information related to how the eProsima team handles the versioning of `fast-cdr`. However, [each release](https://github.com/eProsima/Fast-CDR/releases) is documented in their documentation with the included features and bug fixes.
+
+### Version Stability [1.ii]
+
+The library has version >= 1.0.0.
+
+### Public API Declaration [1.iii]
+    
+The API is declared using Doxygen documentation available to be [generated](https://github.com/eProsima/Fast-CDR/blob/master/utils/doxygen/doxyfile) with the source code. They have it hosted in this [page](https://www.eprosima.com/docs/fast-buffers/0.3.0/html/group___f_a_s_t_c_d_r_a_p_i_r_e_f_e_r_e_n_c_e.html) for the version 0.3.0. However, that is an outdated version, as currently the software version is 1.0.13.
+
+### API Stability Policy [1.iv]
+    
+There is no explicit policy related to API stability. This package should be pinned to a particular Fast-CDR version to be considered high quality.
+
+### ABI Stability Policy [1.v]
+    
+There is no explicit policy related to ABI stability. This package should be pinned to a particular Fast-CDR version to be considered high quality.
+
+### ABI and ABI Stability Within a Released ROS Distribution [1.vi]
+   
+There is no direct correlation between API and ABI stability of the library within ROS distributions. This package should be pinned to a particular Fast-CDR version to be considered high quality.
```

Comment:
"Individual ROS distributions may need to choose specific versions to ensure API and ABI stability within a released ROS distribution"